src/plugin.c

```diff
@@ -253,6 +253,56 @@ static int register_callback (llist_t **list, /* {{{ */
 	return (0);
 } /* }}} int register_callback */
 
+static void log_list_callbacks (llist_t **list, /* {{{ */
+				const char *comment)
+{
+	char *str;
+	int len;
+	llentry_t *le;
+	int i;
+	int n;
+	char **keys;
+
+	n = llist_size(*list);
```

Comment:
I would simplify this function considerably, essentially doing:

```c
INFO ("Available write targets:");
for (le = llist_head (*list), i = 0, len = 0;
    le != NULL;
    le = le->next, i++)
{
  INFO ("* \"%s\"", le->key);
}
```

My reasoning is: This is not printed very often. If it is printed, there will be only a handful of lines, usually less then half a dozen. So all the allocating / prettyprinting is adding more complexity than I feel is justified.